lib/ast.ml

```diff
@@ -33,6 +33,21 @@ and pattern' =
     | PatNil
 [@@deriving sexp_of]
 
+let line_number_of_t = (function
```

Comment:
nits: parens are not needed.

```ocaml
let line_number_of_t = function
| Constant (line, _) -> line
...
| ListNil -> -1

let string_of_t t = ...
```

is valid syntax.

Reply:
I see.